src/shogun/lib/any.h

```diff
@@ -349,9 +356,22 @@ namespace shogun {
 		{
 		}
 
+		template <
+		    typename T,
+		    std::enable_if_t<std::is_same_v<
+		        std::result_of<std::function<T()>>::type, void>>* = nullptr>
+		void on(std::function<T()>* v)
+		{
+			value = std::make_shared<T>((*v)());
+			return on(static_cast<T*>(value.get()));
```

Comment:
@vigsterkr [it ain't much...](https://i.kym-cdn.com/entries/icons/mobile/000/028/021/work.jpg)

Reply:
the rest in this file is just clang-format